src/include/network/routines.rb

```diff
@@ -461,37 +299,27 @@ def DistinguishedName(name, hwdevice)
     # @param [Hash] hwdevice hardware device
     # @return name consisting of vendor and device name
     def DeviceName(hwdevice)
-      hwdevice = deep_copy(hwdevice)
-      delimiter = " " # "\n"; #FIXME: constant
+      device = hwdevice["device"] || ""
+      return device if !device.empty?
 
-      if IsNotEmpty(Ops.get_string(hwdevice, "device", ""))
-        return Ops.get_string(hwdevice, "device", "")
-      end
-
-      model = Ops.get_string(hwdevice, "model", "")
-      return model if model != "" && !model.nil?
+      model = hwdevice["model"] || ""
+      return model if !model.empty?
 
-      vendor = Ops.get_string(hwdevice, "sub_vendor", "")
-      dev = Ops.get_string(hwdevice, "sub_device", "")
+      vendor = hwdevice["sub_vendor"] || ""
+      dev = hwdevice["sub_device"] || ""
 
-      if vendor == "" || dev == ""
-        vendor = Ops.get_string(hwdevice, "vendor", "")
-        dev = Ops.get_string(hwdevice, "device", "")
+      if vendor.empty? || dev.empty?
+        vendor = hwdevice["vendor"] || ""
+        dev = hwdevice["device"] || ""
       end
 
-      if vendor != ""
-        return Ops.add(Ops.add(vendor, delimiter), dev)
-      else
-        return dev
-      end
+      "#{vendor} #{dev}".strip
     end
 
     # Validates given name for use as a nic name in sysconfig. See bnc#784952
     def ValidNicName(name)
       # 16 is the kernel limit on interface name size (IFNAMSIZ)
-      return false if !Builtins.regexpmatch(name, "^[[:alnum:]._:-]{1,15}$")
-
-      true
+      !(name =~ /^[[:alnum:]._:-]{1,15}$/).nil?
```

Comment:
`^` and `$` match LINES, not the entire string, in [Ruby regexes](http://ruby-doc.org/core-2.0.0/Regexp.html#class-Regexp-label-Anchors), so `"eth0\n" `would match. Use `\A` and `\z`.